fenix/app/src/main/java/org/mozilla/fenix/gptintegration/apis/GPTIntegrationApi.kt

```diff
@@ -0,0 +1,49 @@
+package org.mozilla.fenix.gptintegration.apis
+
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.MutableHeaders
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import org.json.JSONArray
+import org.json.JSONObject
+import org.mozilla.fenix.utils.Settings
+
+private const val BASE_API_URL = "https://api.openai.com/"
+
+class GPTIntegrationApi(
+    settings: Settings,
+    private val client: Client,
+): GPTIntegrationApiInterface {
+
+    private val apiKey = settings.gptApiKey
+
+    override suspend fun fetch(body: String?): Response {
+        val jsonObject = JSONObject()
+        val jsonArray = JSONArray()
+        val jsonObjectSubItem = JSONObject()
+        jsonObjectSubItem.put("role", "user")
+        jsonObjectSubItem.put("content",
+         "Summarize the following text: $body"
+        )
+        jsonArray.put(jsonObjectSubItem)
+        jsonObject.put("model", "gpt-3.5-turbo")
+        jsonObject.put("max_tokens", 1028)
```

Comment:
Currently, OpenAI has set a limitation of # tokens per prompt for any available model, the chosen model is the top 2. In a production app, I would bring these details to the team and try to understand the use case/necessities of this feature, in order to implement accordingly.  More details about tokens limitation [here](https://openai.com/pricing).